src/annotator/integrations/html.js

```diff
@@ -38,9 +48,97 @@ export class HTMLIntegration {
+      return false;
+    }
+
+    if (layout.expanded) {
+      this._activateSideBySide(layout.width);
+      return true;
+    } else {
+      this._deactivateSideBySide();
+      return false;
+    }
+  }
+
+  /**
+   * Resize the document content after side-by-side mode is activated.
+   *
+   * @param {number} sidebarWidth
+   */
+  _activateSideBySide(sidebarWidth) {
+    // When side-by-side mode is activated, what we want to achieve is that the
+    // main content of the page is fully visible alongside the sidebar, with
+    // as much space given to the main content as possible. A challenge is that
+    // we don't know how the page will respond to reducing the width of the body.
+    //
+    // - The content might have margins which automatically get reduced as the
+    //   available width is reduced. For example a blog post with a fixed-width
+    //   article in the middle and `margin: auto` for both margins.
+    //
+    //   In this scenario we'd want to reduce the document width by the full
+    //   width of the sidebar.
+    //
+    // - There might be sidebars to the left and/or right of the main content
+    //   which cause the main content to be squashed when the width is reduced.
+    //   For example a news website with a column of ads on the right.
+    //
+    //   In this scenario we'd want to not reduce the document width or reduce
+    //   it by a smaller amount and let the Hypothesis sidebar cover up the
+    //   document's sidebar, leaving as much space as possible to the content.
+    //
+    // Therefore what we do is to initially reduce the width of the document by
+    // the full width of the sidebar, then we use heuristics to analyze the
+    // resulting page layout and determine whether there is significant "free space"
+    // (ie. anything that is not the main content of the document, such as ads or
+    // links to related stories) to the right of the main content. If there is,
+    // we make the document wider again to allow more space for the main content.
+    //
+    // These heuristics assume a typical "article" page with one central block
+    // of content. If we can't find the "main content" then we just assume that
+    // everything on the page is potentially content that the user might want
+    // to annotate and so try to keep it all visible.
+    const padding = 10;
+    const rightMargin = sidebarWidth + padding;
+
+    // nb. Adjusting the body size this way relies on the page not setting a
+    // width on the body. For sites that do this won't work.
+    document.body.style.marginRight = `${rightMargin}px`;
+
+    const contentArea = guessMainContentArea(document.body);
+    if (contentArea) {
+      // Check if we can give the main content more space by letting the
+      // sidebar overlap stuff in the document to the right of the main content.
+      const freeSpace = Math.max(
+        0,
+        window.innerWidth - sidebarWidth - contentArea.right
+      );
```

Comment:
I believe there may be a small computational bug here in that this is not taking the added `padding` into account. `freeSpace` ends up being `10` (i.e. `padding` value) when I test it on local HTML documents (e.g. enable side-by-side for HTML and then load up _Anna Karenina_: http://localhost:3000/document/tolstoy). Thus the margin gets re-computed and we lose the right-padding and the text runs up against the toolbar:

<img width="107" alt="Screen Shot 2022-02-24 at 8 30 49 AM" src="https://user-images.githubusercontent.com/439947/155534904-7c09d8fc-fc2e-43bb-b695-ba160deedd5f.png">

If I update this computation to take `padding` into account

```suggestion
      const freeSpace = Math.max(
        0,
        window.innerWidth - sidebarWidth - padding - contentArea.right
      );
```
The result seems to be what I believe is intended?

<img width="80" alt="Screen Shot 2022-02-24 at 8 37 19 AM" src="https://user-images.githubusercontent.com/439947/155534938-81e7d901-00d4-4bf4-96c4-d3d92192dc35.png">

Reply:
You are correct. The `sidebarWidth` value here should be replaced by `rightMargin`, which is the sidebar width plus the padding.